packages/react/src/ReactServerContext.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {REACT_PROVIDER_TYPE, REACT_CONTEXT_TYPE} from 'shared/ReactSymbols';
+
+import type {
+  ReactServerContext,
+  ServerContextJSONValue,
+} from 'shared/ReactTypes';
+
+const globalRegistry: {
+  [globalName: string]: ReactServerContext<any>,
+} = {};
+
+export function createServerContext<T: ServerContextJSONValue>(
+  globalName: string,
+  defaultValue: T,
+): ReactServerContext<T> {
+  if (globalRegistry[globalName]) {
+    throw new Error('ServerContext in that name already exists');
```

Comment:
If you create this on the server, then pass it to the client. Then load a file on the client that does something like:

```js
import MyContext from "MyServerContext.js";
...
useContext(MyContext);
```
and MyServerContext.js is:
```
export default createServerContext(...);
```

Then this will error because that module loaded later.

Reply:
Oh yeah I guess I'll skip this check and use `getOrCreate` instead. I was trying to guard against it being created multiple times with different defaults and then users seeing  weird behavior.

Reply:
Also in that case where the server sends it first, I suppose we should also send the default value from the server? But probably not everytime its provided so instead we can send a row to create it before the model is processed.